rts/lra/lra-test/lra-test-basic/src/test/java/io/narayana/lra/arquillian/resource/LRAParticipantWithStatusURI.java

```diff
@@ -0,0 +1,125 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.arquillian.resource;
+
+import org.eclipse.microprofile.lra.annotation.Compensate;
+import org.eclipse.microprofile.lra.annotation.Complete;
+import org.eclipse.microprofile.lra.annotation.Forget;
+import org.eclipse.microprofile.lra.annotation.ParticipantStatus;
+import org.eclipse.microprofile.lra.annotation.Status;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_CONTEXT_HEADER;
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_RECOVERY_HEADER;
+
+@ApplicationScoped
+@Path(LRAParticipantWithStatusURI.LRA_PARTICIPANT_PATH)
+public class LRAParticipantWithStatusURI {
+    public static final String LRA_PARTICIPANT_PATH = "participant-with-status-reporting";
+    public static final String TRANSACTIONAL_CLOSE_PATH = "close-work";
+    public static final String TRANSACTIONAL_CANCEL_PATH = "cancel-work";
+    public static final String FORGET_COUNT_PATH = "forget-count";
+
+    private static final AtomicInteger forgetCount = new AtomicInteger(0);
+    private static final AtomicInteger compensateCount = new AtomicInteger(0);
+    private static final AtomicInteger completeCount = new AtomicInteger(0);
+
+    @GET
+    @Path(TRANSACTIONAL_CLOSE_PATH)
+    @LRA
+    public Response closeLRA(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+        return Response.ok(lraId.toASCIIString()).build();
+    }
+
+    @GET
+    @Path(TRANSACTIONAL_CANCEL_PATH)
+    @LRA(cancelOn = Response.Status.INTERNAL_SERVER_ERROR)
+    public Response abortLRA(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+        return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(lraId.toASCIIString()).build();
+    }
+
+    @GET
+    @Path(FORGET_COUNT_PATH)
+    public int getForgetCount() {
+        return forgetCount.get();
+    }
+
+    @PUT
+    @Path("compensate")
+    @Produces(MediaType.APPLICATION_JSON)
```

Comment:
it produces Enums name, so it should be probably text/plain.

Reply:
Okay

Reply:
Fixed with the last commit